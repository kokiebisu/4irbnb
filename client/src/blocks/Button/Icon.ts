import styled from 'styled-components';

const Icon = styled.div``;

export default Icon;
