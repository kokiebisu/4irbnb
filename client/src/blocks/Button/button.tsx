import React from 'react';
import styled, { css } from 'styled-components';

interface Props {
  children: string;
  onPress: () => void;
  className?: string;
  size: number;
}

interface ButtonProps {
  size: number;
}

const Button = styled.button<ButtonProps>`
  font-family: 'Airbnb-Cereal';
  border-radius: 5px;
  font-size: 18px;
  border: none;

  ${({ size }) =>
    (size === 3 &&
      css`
        padding: 10px;
      `) ||
    (size === 5 &&
      css`
        padding: 15px;
      `) ||
    (size === 7 &&
      css`
        padding: 20px;
      `)}
`;

export const BaseButton: React.FC<Props> = ({
  className,
  onPress,
  children,
  size,
}) => {
  return (
    <Button size={size} className={className} onClick={onPress}>
      {children}
    </Button>
  );
};
