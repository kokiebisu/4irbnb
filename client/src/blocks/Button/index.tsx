import React from 'react';
import styled, { css } from 'styled-components';
import { rgba, lighten } from 'polished';

interface Props {
  styles: any;
}

const BaseButton: React.FC<Props> = ({ styles, children }) => {
  return <button {...css(styles.button)}>{children}</button>;
};

export default BaseButton;
