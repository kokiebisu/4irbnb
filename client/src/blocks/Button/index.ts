import styled from 'styled-components';
import { rgba, lighten } from 'polished';

// Particles
import Text from './Text';
import Icon from './Icon';

const Button: any = styled.button`
  background-color: #7d7d7d;
  border-radius: 2px;
  border: solid 1px transparent;
  color: #fff;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.5rem 1.5rem;
  outline: none;
  transition: all 200ms ease;

  &:hover,
  &:focus {
    background-color: ${lighten(0.05, '#7D7D7D')};
    box-shadow: 0 8px 8px 0 ${rgba('#646464', 0.4)};
  }
`;

Button.Text = Text;
Button.Icon = Icon;

export default Button;
