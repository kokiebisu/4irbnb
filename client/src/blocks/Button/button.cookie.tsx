import React from 'react';
import { css } from 'styled-components';
import theme from 'styled-theming';

// element
import { BaseButton } from '../../elements/Button';
import { Text } from '../../elements/Text';
import { colorpallete } from '../../styles/colorpallete';

// type
import { ButtonProps } from './button';

export default ({ inverse, children, ...props }: ButtonProps) => {
  return (
    <BaseButton
      styles={inverse ? extend.inverse.button : extend.normal.button}
      {...props}>
      <Text styles={inverse ? extend.inverse.text : extend.normal.text}>
        {children}
      </Text>
    </BaseButton>
  );
};

const primary = theme('mode', {
  light: colorpallete.bluegreen__5,
});

const secondary = theme('mode', {
  light: colorpallete.white,
});

const common = css`
  padding: 15px 0;
  font-size: 12px;
  font-weight: 700;
  min-width: 125px;
`;

const extend = {
  normal: {
    button: css`
      ${common}
      background-color: ${primary};
    `,
    text: css`
      color: ${secondary};
    `,
  },
  inverse: {
    button: css`
      ${common}
      background-color: ${secondary}; 
      border: 1px solid ${primary};
    `,
    text: css`
      color: ${primary};
    `,
  },
};
