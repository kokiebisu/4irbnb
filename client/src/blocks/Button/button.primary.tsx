import { BaseButton } from './button';
import styled, { css } from 'styled-components';

export const PrimaryButton = styled(BaseButton)`
  background: -webkit-linear-gradient(
    right,
    rgb(230, 30, 77) 0%,
    rgb(227, 28, 95) 50%,
    rgb(215, 4, 102) 100%
  );
  border: none;
  color: white;
`;
