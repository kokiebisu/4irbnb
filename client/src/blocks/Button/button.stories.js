import React from 'react';
import { Button } from './button';

export default {
  title: 'Design System/Blocks/Buttons',
};

export const BaseButton = () => <Button>Hello</Button>;
