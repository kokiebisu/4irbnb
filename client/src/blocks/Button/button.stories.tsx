import React from 'react';

// Component
import { BaseButton } from './button';
import { PrimaryButton } from './button.primary';

export default {
  title: 'Design System/Blocks/Buttons',
};

export const Base = () => {
  return (
    <BaseButton onPress={() => console.log('base')}>Base BUtton</BaseButton>
  );
};

export const Primary = () => {
  return (
    <PrimaryButton onPress={() => console.log('primary')}>
      Primary Button
    </PrimaryButton>
  );
};
