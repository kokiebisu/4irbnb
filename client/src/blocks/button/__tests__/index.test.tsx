import React from 'react';
import { Button } from '../../../elements/Button';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Base Button', () => {
  it('renders correctly', () => {
    const testingData = {
      children: 'test',
      action: () => console.log('test'),
    };

    const tree = renderer
      .create(<Button sm onPress={testingData.action} {...testingData} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
