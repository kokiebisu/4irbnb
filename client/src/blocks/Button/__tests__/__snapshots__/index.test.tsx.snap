// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base Button renders correctly 1`] = `
.c0 {
  background-color: transparent;
  border: 1px solid red;
  color: red;
}

<button
  className="c0"
  onClick={[Function]}
>
  button
</button>
`;
